### PR TITLE
MULE-19499: Support new SDK API OAuth annotation package

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OAuthConnectionProviderModelValidator.java [hunk 1]
```diff
@@ -46,27 +53,62 @@ protected void onConnectionProvider(ConnectionProviderModel model) {
           return;
         }
 
-        AuthorizationCode authCode = implementingType.getAnnotation(AuthorizationCode.class);
-        ClientCredentials clientCredentials = implementingType.getAnnotation(ClientCredentials.class);
+        Optional<OAuthModelProperty> optionalOAuthModelProperty = model.getModelProperty(OAuthModelProperty.class);
 
-        if (authCode != null && clientCredentials != null) {
+        boolean supportsAuthCode = optionalOAuthModelProperty.map(oAuthModelProperty -> oAuthModelProperty.getGrantTypes()
```
Comment (dabreu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OAuthConnectionProviderModelValidator.java:58]: maybe this code could be extracted to a function and used to calculate supportsAuthCode and supportsClientCredentials? that might reduce the whole method length increasing readability.

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaConnectionProviderModelParser.java [hunk 2]
```diff
@@ -141,23 +152,69 @@ public boolean isExcludedFromConnectivitySchema() {
...
-                                                                      a.includeRedirectUriInRefreshTokenRequest())));
-
-    element.getAnnotation(ClientCredentials.class).ifPresent(a -> grantTypes.add(new ClientCredentialsGrantType(a.tokenUrl(),
-                                                                                                                a.accessTokenExpr(),
-                                                                                                                a.expirationExpr(),
-                                                                                                                a.defaultScopes(),
-                                                                                                                a.credentialsPlacement()))
-
-    );
+    mapReduceSingleAnnotation(element, CONNECTION_PROVIDER_NAME, element.getName(), AuthorizationCode.class,
```
Comment (dabreu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaConnectionProviderModelParser.java:155]: would it be possible to add a couple methods for creating the AuthorizationCodeGrantType and the ClientCredentialsGrantType in order to make the getOAuthModelProperty() method more modular/readable? what do you think?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OAuthConnectionProviderModelValidator.java [hunk 3]
```diff
@@ -46,27 +53,62 @@ protected void onConnectionProvider(ConnectionProviderModel model) {
...
-        if (authCode != null) {
+        if (supportsAuthCode) {
           validateStateField(implementingType, AuthorizationCodeState.class, "authorization code");
         }
 
-        if (clientCredentials != null) {
+        if (supportsClientCredentials) {
           validateStateField(implementingType, ClientCredentialsState.class, "client credentials");
         }
+
```
Comment (dabreu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OAuthConnectionProviderModelValidator.java:82]: this block of code might also be moved to a separate function to reveal its intention and enhance readability, what do you think?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaConnectionProviderModelParser.java (hunk 2 above)
Comment (ndinu) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaConnectionProviderModelParser.java:155]: Totally, I will do it.

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OAuthConnectionProviderModelValidator.java (hunk 1 above)
Comment (ndinu) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OAuthConnectionProviderModelValidator.java:58]: 👍

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OAuthConnectionProviderModelValidator.java (hunk 3 above)
Comment (ndinu) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OAuthConnectionProviderModelValidator.java:82]: agree

**Review by dabreu — APPROVED** [review]

**Pull request merged**